Readmes/choose camera controller to control annotation sublayer visibility (#717)
Esri/arcgis-runtime-samples-android: kotlin/configure-subnetwork-trace/README.md

1. Populate the traversability scope list with the enum values from `UtilityTraversabilityScope`.
2. Create and load a `UtilityNetwork` with a feature service URL, then get an asset type and a tier by their names.
3. Populate the choice list for the comparison source with the non-system defined `Definition.networkAttributes`.  Populate the choice list for the comparison operator with the enum values from `UtilityAttributeComparisonOperator`.
4. Create a `UtilityElement` from this asset type to use as the starting location for the trace.
5. Update the selected barrier expression and the checked options in the UI using this tier's `TraceConfiguration`.

> 6. When 'Network Attribute' is selected, if its `Domain` is a `CodedValueDomain`, populate the choice list for the comparison value with its `CodedValues`.  Otherwise, display a free-form textbox for entering an attribute value.

Reviewer: I think we could improve this instruction since there is no class called just "Domain" used in this sample
```suggestion
6. When 'Network Attribute' is selected, check whether it has a `CodedValueDomain`, and if so, populate the choice list for the comparison value with its `CodedValues`.  Otherwise, display a free-form textbox for entering an attribute value.
```

> 7. When 'Add' is clicked, create a new `UtilityNetworkAttributeComparison` using the selected comparison source, operator, and selected or typed value. Use the selected source's `NetworkAttribute.DataType` to convert the comparison value to the correct data type.

Reviewer: Just noticed we're not referring to the class directly here
```suggestion
7. When 'Add' is clicked, create a new `UtilityNetworkAttributeComparison` using the selected comparison source, operator, and selected or typed value. Use the selected source's `UtilityNetworkAttribute.DataType` to convert the comparison value to the correct data type.
```

> 8. If `Traversability.barriers` is not empty, create a `UtilityTraceOrCondition` with the existing `Barriers` and the new comparison from step 8.

Reviewer: we also don't have a class called `Barriers` so let's reformat this
```suggestion
8. If `Traversability.barriers` is not empty, create a `UtilityTraceOrCondition` with the existing barriers and the new comparison from step 8.
```

> 9. When 'Trace' is clicked, create `UtilityTraceParameters` passing in `UtilityTraceType.SUBNETWORK` and the default starting location.  Set its `TraceConfiguration` with the modified options, selections, and expression; then run a `UtilityNetwork.traceAsync(...)`.

Reviewer: same here, our API class is `UtilityTraceConfiguration`
```suggestion
9. When 'Trace' is clicked, create `UtilityTraceParameters` passing in `UtilityTraceType.SUBNETWORK` and the default starting location.  Set its `UtilityTraceConfiguration` with the modified options, selections, and expression; then run a `UtilityNetwork.traceAsync(...)`.
```

1. Populate the traversability scope list with the enum values from `UtilityTraversabilityScope`.
2. Create and load a `UtilityNetwork` with a feature service URL, then get an asset type and a tier by their names.
3. Populate the choice list for the comparison source with the non-system defined `UtilityNetworkDefinition.networkAttributes`.  Populate the choice list for the comparison operator with the enum values from `UtilityAttributeComparisonOperator`.
4. Create a `UtilityElement` from this asset type to use as the starting location for the trace.
5. Update the selected barrier expression and the checked options in the UI using this tier's `UtilityTraceConfiguration`.

> 6. When 'Network Attribute' is selected, if its `Domain` is a `CodedValueDomain`, populate the choice list for the comparison value with its `CodedValues`.  Otherwise, display a free-form textbox for entering an attribute value.

Reviewer: whoops, looks like `Domain` snuck back in again. Remember this isn't part of our API, I think you had corrected this in an earlier commit. Like this:

> When 'Network Attribute' is selected, check whether it has a `CodedValueDomain`, and if so...

> 7. When 'Add' is clicked, create a new `UtilityNetworkAttributeComparison` using the selected comparison source, operator, and selected or typed value. Use the selected source's `NetworkAttribute.DataType` to convert the comparison value to the correct data type.

Reviewer: looks like this got reverted to, should be `UtilityNetworkAttribute.DataType`

8. If the utility trace configuration's `UtilityTraversability.barriers` is not empty, create a `UtilityTraceOrCondition` with the existing barriers and the new comparison from step 7.

> 9. When 'Trace' is clicked, create `UtilityTraceParameters` passing in `UtilityTraceType.SUBNETWORK` and the default starting location.  Set its `TraceConfiguration` with the modified options, selections, and expression; then run a `UtilityNetwork.traceAsync(...)`.

Reviewer: this should be `UtilityTraceConfiguration`, looks like it got reverted

> 5. Update the selected barrier expression and the checked options in the UI using this tier's `TraceConfiguration`.

Reviewer: oops sorry I should have spotted this before, this should be `UtilityTraceConfiguration` (all of our UtilityNetwork API classes have the prefix of `Utility`, just for future note!)

Reviewer: sorry to add more changes on but since we're editing this readme anyway, I also noticed on line 23 there is a reference to `Definition.networkAttributes`. Could you change this to `UtilityNetworkDefinition` please?

Author: Thanks, will keep checking for these!

6. When 'Network Attribute' is selected, check whether it has a `CodedValueDomain`, and if so, populate the choice list for the comparison value with its `CodedValues`.  Otherwise, display a free-form textbox for entering an attribute value.
7. When 'Add' is clicked, create a new `UtilityNetworkAttributeComparison` using the selected comparison source, operator, and selected or typed value. Use the selected source's `UtilityNetworkAttribute.DataType` to convert the comparison value to the correct data type.

> 8. If `Traversability.barriers` is not empty, create a `UtilityTraceOrCondition` with the existing barriers and the new comparison from step 8.

Reviewer: just a clarification suggestion to add here: 
```suggestion
8. If the utility trace configuration's `UtilityTraversability.barriers` is not empty, create a `UtilityTraceOrCondition` with the existing barriers and the new comparison from step 8.
```

## About the data

> The [Naperville electrical](https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer)  network feature service, hosted on ArcGIS Online, contains a utility network used to run the subnetwork-based trace shown in this sample.

Reviewer: extra space snuck in here
```suggestion
The [Naperville electrical](https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer) network feature service, hosted on ArcGIS Online, contains a utility network used to run the subnetwork-based trace shown in this sample.
```
